stablehlo/reference/Tensor.h

```diff
@@ -70,8 +70,6 @@ class Tensor {
   /// \name Constructors
   /// @{
   Tensor();
-  // TODO: Remove the need in having this constructor.
-  explicit Tensor(Type type) : Tensor(type.cast<RankedTensorType>()) {}
```

Comment:
Hmm that's a lot of casts. How about we replace `RankedTensorType` with `TensorType` for now? This way, we will be able to remove all the casts.